### ISSUE TITLE
[bug] Text shapes deleted by erasing text break the undo/redo stack

### ISSUE THREAD

**Issue by Claydelas** [tldraw/tldraw#515]
### Here's how to reproduce:
![demo](https://i.imgur.com/zV22nW4.gif)

1. Add a text shape
2. Add text to it
3. Delete the text
4. Undo

Looks like this is similar to #502 and #489 in that the exception arises in useShapeTree.

**Comment by Claydelas** [comment]
The undo-restored text shape doesn't seem to keep most of its previous TLShape properties.
Before undo:
```json
{
    "b7ceb535-7f81-44bb-2a3f-0aefd446cf07": {
        "id": "b7ceb535-7f81-44bb-2a3f-0aefd446cf07",
        "type": "text",
        "name": "Text",
        "parentId": "page",
        "childIndex": 1,
        "point": [
            401,
            450
        ],
        "rotation": 0,
        "text": " ",
        "style": {
            "color": "black",
            "size": "small",
            "isFilled": false,
            "dash": "draw",
            "scale": 1,
            "font": "script",
            "textAlign": "start"
        }
    }
}
```
After undo:
```json
{
    "id": "b7ceb535-7f81-44bb-2a3f-0aefd446cf07",
    "point": [
        401,
        450
    ],
    "text": "a"
}
```

**Comment by steveruizok** [comment]
Great fix!

**Comment by steveruizok** [comment]
Closed by #516. Great find.
